plugins/inventory/vmware_host_inventory.py

```diff
@@ -0,0 +1,421 @@
+
+        self.pyv = BaseVMwareInventory(
+            hostname=self.get_option('hostname'),
+            username=self.get_option('username'),
+            password=password,
+            port=self.get_option('port'),
+            with_tags=self.get_option('with_tags'),
+            validate_certs=self.get_option('validate_certs')
+        )
+
+        self.pyv.do_login()
+
+        if cache:
+            cache = self.get_option('cache')
+
+        update_cache = False
+        if cache:
+            try:
+                cacheable_results = self._cache[cache_key]
+            except KeyError:
+                update_cache = True
+
+        if cache and not update_cache:
+            self._populate_from_cache(cacheable_results)
+        else:
+            cacheable_results = self._populate_from_source()
+
+        if update_cache or (not cache and self.get_option('cache')):
+            self._cache[cache_key] = cacheable_results
+
+    def _populate_from_cache(self, cache_data):
+        """
+        Populate cache using source data
+
+        """
+        for host, host_properties in cache_data.items():
+            self._populate_host_properties(host_properties, host)
+
+    def _populate_from_source(self):
+        """
+        Populate inventory data from direct source
+
+        """
+        hostvars = {}
+        strict = self.get_option('strict')
+
+        host_properties = self.get_option('properties')
+        if not isinstance(host_properties, list):
+            host_properties = [host_properties]
+
+        if len(host_properties) == 0:
+            host_properties = ['name']
+
+        if 'all' in host_properties:
+            query_props = None
+            host_properties.remove('all')
+        else:
+            if 'runtime.connectionState' not in host_properties:
+                host_properties.append('runtime.connectionState')
+            query_props = [x for x in host_properties if x != "customValue"]
+
+        objects = self.pyv.get_managed_objects_properties(
+            vim_type=vim.HostSystem,
+            properties=query_props,
+            resources=self.get_option('resources'),
+            strict=strict,
+        )
+
+        tags_info = dict()
+        if self.pyv.with_tags:
+            tag_svc = self.pyv.rest_content.tagging.Tag
+            cat_svc = self.pyv.rest_content.tagging.Category
+
+            tags = tag_svc.list()
+            for tag in tags:
+                tag_obj = tag_svc.get(tag)
+                tags_info[tag_obj.id] = (tag_obj.name, cat_svc.get(tag_obj.category_id).name)
+
+        hostnames = self.get_option('hostnames')
+
+        for host_obj in objects:
+            properties = dict()
+            for host_obj_property in host_obj.propSet:
+                properties[host_obj_property.name] = host_obj_property.val
+
+            if (properties.get('runtime.connectionState') or properties['runtime'].connectionState) in ('orphaned', 'inaccessible'):
```

Comment:
It looks like maybe this script was converted from the VM inventory so it inherited some some things that relate to VMs but not hosts. 'orphaned' and 'inaccessible' are valid states only for VMs, so if you want to keep this check, consider changing this to the host equivalents. It should be 'disconnected' and 'notresponding' but I did not check this.

Reply:
Yes, you are right. I will check and update these states accordingly. Thanks for testing and providing this feedback.

Reply:
HostSystemConnectionState = https://vdc-download.vmware.com/vmwb-repository/dcr-public/bd170cdc-e03a-4701-b468-bcfba05da6b2/946a76df-5276-46fa-a10d-06f0f7d4078e/doc/vim.HostSystem.ConnectionState.html